### DISCUSSION TITLE
Could not read on Ubuntu from Github repository over SSH

### DISCUSSION THREAD

**Discussion by ghost** [community/community#38362] [Code Security] [answered]
I am attempting to push my local repository to GitHub using Ubuntu CLI and SSH. My SSH key has been installed on Github and in ssh-agent on my Ubuntu machine. I have verified that ssh -T git @github works, implying that I can connect to Github. I get the message "Could not read from remote repository" and don´t know where to start debugging. The remotes on the local machine seem to have been set with the correct URL.

**Comment by airtower-luna** [comment]
Please post the full output of your push command, otherwise it's hard to guess what may be going on. You can censor things like a private repo name, but the full output is critical.

**Reply by ghost** [reply]
Here is the exact text:
jonathan@JFG-dev:~/ProgrammingFreedom/General$ git push origin main
github@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.

**Comment by airtower-luna** [comment, marked as answer]
Looks like something is wrong with the URL:

`github@github.com: Permission denied (publickey).`

The SSH URL should start with `git@github.com`.
